qutebrowser/completion/models/listcategory.py

```diff
@@ -0,0 +1,66 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2017 Ryan Roden-Corrent (rcorre) <ryan@rcorre.net>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The base completion model for completion in the command line.
+
+Module attributes:
+    Role: An enum of user defined model roles.
+"""
+
+from PyQt5.QtCore import Qt
+from PyQt5.QtGui import QStandardItem, QStandardItemModel
+
+
+class ListCategory(QStandardItemModel):
+
+    """Expose a list of items as a category for the CompletionModel."""
+
+    def __init__(self, name, items, parent=None):
+        super().__init__(parent)
+        self.name = name
+        # self.setColumnCount(3) TODO needed?
+        # TODO: batch insert?
```

Comment:
for some reason I was thinking `insertRows` was more efficient (so it only calls [`beginInsertRows`](http://doc.qt.io/qt-5/qabstractitemmodel.html#beginInsertRows) once, but looking at it again I think that's just to prevent firing a bunch of changed signals, which aren't hooked up at this point.